### PR TITLE
Fix select with prompt still being valid #16

### PR DESCRIPTION
**Pull request by srsgores**
#16

### CODE REVIEW

File: tests/integration/components/select-list-test.js [hunk 1]
```diff
@@ -183,3 +183,15 @@ test('binds properly to `title` attribute', function(assert) {
...
+test('remains invalid when prompt is selected', function(assert) {
+  var options = [];
+  this.set('options', options);
+  this.set('prompt', 'First prompt');
+
+  this.render(hbs`
+    {{select-list content=options value=selection prompt=prompt}}
+  `);
+
+  assert.equal(this.$('select option:selected')[0].validity.valid, false);
```
Comment (begedin) [inline tests/integration/components/select-list-test.js]: This fails in chrome (or phantomjs) and only seems to be working in Firefox. If you have another idea on how to check for valid state, go with it.

Alternatively, if there is no other way, we can just check to make sure the selected value of the select element itself is set to nothing.

```
assert.equal(this.$('select').val(), null);
```

That should be good enough.

**Comment by begedin** [comment]
Two things need fixing
1. The test you added is failing in chrome or  phantom js
2. The changes break an existing test (`unit/it renders prompt correctly`). Replace the existing assertion:

```
assert.equal(this.$('option').attr('value'), undefined, 'The prompt option has no value assigned to it');
```

With

```
assert.equal(this.$('option').attr('value'), '', 'The prompt option has no value assigned to it');
```

**Comment by srsgores** [comment]
@begedin, I gave you push access to my fork.  Feel free to make the changes.

File: tests/integration/components/select-list-test.js (hunk 1 above)
Comment (srsgores) [inline tests/integration/components/select-list-test.js]: Not good enough; `validity` is what the browser uses to determine whether the form is valid.  Especially when you only have 1 element in your form, and that is a `select`, this makes the difference between the form submitting and not.  Simply testing for `value=""` isn't enough; we need to test that the `<select>` element is invalid.

Comment (begedin) [inline tests/integration/components/select-list-test.js]: True, but we don't have that api in other browsers, so the closest thing to do in order to test it is to set the `required` attribute and then check if the select element has a value. Perhaps testing for $('select').val() being truthy is a bit closer, but we can't make a validity test that would pass in either chrome or phantom js (which is what our two test runners are).

Comment (srsgores) [inline tests/integration/components/select-list-test.js]: Hmm, so I guess the best compromise ATM is `(this.$("select").val() === null || this.$("select")[0].validity === false)`?  I guess I'm ok with that, so long as it passes.

**Comment by srsgores** [comment]
@begedin, what's the status on this?

**Comment by begedin** [comment]
@srsgores I'm sorry. I'll finalize and merge it this weekend. I've been extremely busy for the past few days.

**Pull request merged**